File: cpp/09proc/Proc39.cpp
#include <iostream>
#include <cmath>
using namespace std;

double Power1(double A, double B);
double Power2(double A, int N);
double Power3(double A, double B);

int main()
{
	//Task("Proc39");
	double P, N;
	cin >> P;
	for (int i = 1; i < 4; i++)
	{
		cin >> N;
		cout << Power3(N, P);
	}
	
	return 0;
}

double Power3(double A, double B)
{
	int N = B;
	
	return ( N == B ? Power2(A, N) : Power1(A, B) );
}

double Power1(double A, double B)
{
	if (A <= 0)
		return 0;
	
	return exp(B * log(A));
}

double Power2(double A, int N)
{
	if (N == 0)
		return 1;
	
	bool isNegative = false;
	if (N < 0)
	{
		isNegative = true;
		N *= -1;
	}
	double degreeN = 1;
	for (int i = 1; i <= N; i++)
		degreeN *= A;

	if (isNegative)
		return 1 / degreeN;
	return degreeN;
}

File: php/09proc/proc39.php
<?php
function Power1($A, $B){
        $nat = exp($B * log($A));
        return $nat;
}
function Power2($A, $N){
        $dar = 1;
        for($co = 1; $co <= $N; $co++)
                $dar *= $A;
        return $dar;
}
function Power3($A, $B){
        $but = intval($B);
        if($but == $B)
                $nat = Power2($A, $B);
        else
                $nat = Power1($A, $B);
        return $nat;
}
$A = mt_rand(-20, 20);
$B = mt_rand(-20, 20);
$C = mt_rand(-20, 20);
$P = mt_rand(-10, 10);
echo "A = ".$A;
echo "<br>B = ".$B;
echo "<br>C = ".$C;
echo "<br>P = ".$P;
echo "<br>A^P = ".Power3($A, $P);
echo "<br>B^P = ".Power3($B, $P);
echo "<br>C^P = ".Power3($C, $P);
?>